### ISSUE TITLE
Different learning rate per layer

### ISSUE THREAD

**Issue by Kalkasas** [ppwwyyxx/tensorpack#10]
For tensorflow you can use multiple optimizers to achieve different learning rates per layer.
Is there support for this in tensorpack or do I have to write a custom trainer?

**Comment by ppwwyyxx** [comment]
Yes, you can specify different learning rate for each parameter, by scaling its gradient. You can overwrite the `get_gradient_processor()` method in your Model with something like:
```python
    def get_gradient_processor(self):
        return [ScaleGradient([('conv.*/W', 0.5), ('fc.*', 2)])]
```
ScaleGradient takes a list of `(regex, multiplier)` tuple.

**Comment by Kalkasas** [comment]
Exactly what I needed. Thank you!

**Comment by chenchen718** [comment]
I couldn't find any function as `get_gradient_processor()` in the base class `ModelDesc`. I tried adding this function into my Model, but it seemed that this function was never called.

**Comment by ppwwyyxx** [comment]
This usage was deprecated one year ago. Now gradient processors can be added onto optimizer with http://tensorpack.readthedocs.io/en/latest/modules/tfutils.html#tensorpack.tfutils.optimizer.apply_grad_processors .

For example: https://github.com/ppwwyyxx/tensorpack/blob/9972b15036c851ce82e63721cfe0905e68115475/examples/SpatialTransformer/mnist-addition.py#L90-L96
